lib/ansible/modules/cloud/amazon/ec2_tag.py

```diff
@@ -171,17 +188,20 @@ def main():
         if not tags:
             module.fail_json(msg="tags argument is required when state is absent")
         for (key, value) in set(tags.items()):
-            if (key, value) not in set(tagdict.items()):
+            if (key, value) not in set(tagdict.items()) and not purge_tags:
                 baddict[key] = value
                 if set(baddict) == set(tags):
                     module.exit_json(msg="Nothing to remove here. Move along.", changed=False)
         for (key, value) in set(tags.items()):
             if (key, value) in set(tagdict.items()):
                 dictremove[key] = value
+        if purge_tags:
```

Comment:
Yes - this looks good. You do need to put these back into `dict` format since `delete_tags` needs a dictionary format.